src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+    }
+
+    /**
+     * Reads user input and separates the command being invoked from its arguments.
+     * @return a String array of the form {command, args}
+     */
+    private static String[] getUserInput() {
+        System.out.print(INPUT_PROMPT);
+        String userInput = in.nextLine();
+        System.out.println(HORIZONTAL_SEPARATOR);
+        String[] userInputTokenized = userInput.trim().split(" ", 2);
+        if (userInputTokenized.length == 2) {
+            return userInputTokenized;
+        }
+        return new String[] {userInputTokenized[0], ""};
+    }
+
+    /**
+     * Generates an error message for an invalid command.
+     * @param usageExample string showing the correct command format
+     * @return generated error message
+     */
+    private static String invalidCommandError(String command, String usageExample) {
+        return String.format(MESSAGE_INCORRECT_COMMAND_FORMAT, command, usageExample);
+    }
+
+    /**
+     * Attempts to add the specified task to the task list.
+     * @param t task to be added
+     * @param successMessage message to be displayed on success
+     */
+    private static String addTask(Task t, String successMessage) {
+        if (numSavedTasks < MAX_TASKS) {
+            savedTasks[numSavedTasks] = t;
+            numSavedTasks++;
+            return String.format(successMessage, t);
+        }
+        return MESSAGE_TOO_MANY_TASKS;
+    }
+
+    /**
+     * Invokes the correct command function based on the command string inputted by the user.
+     */
+    private static String parseCommand(String[] input) {
+        final String command = input[0];
+        final String args = input[1];
+        switch (command) {
+        case COMMAND_BYE:
+            return byeCommand(args);
+        case COMMAND_LIST:
+            return listCommand(args);
+        case COMMAND_TODO:
+            return todoCommand(args);
+        case COMMAND_DEADLINE:
+            return deadlineCommand(args);
+        case COMMAND_EVENT:
+            return eventCommand(args);
+        case COMMAND_MARK:
+            return markCommand(args);
+        case COMMAND_UNMARK:
+            return unmarkCommand(args);
+        default:
+            return MESSAGE_UNKNOWN_COMMAND;
+        }
+    }
+
+    /**
+     * Request to exit the program.
+     */
+    private static String byeCommand(String args) {
+        if (args.equals("")) {
+            isExitRequested = true;
+            return MESSAGE_GOODBYE;
+        }
+        return invalidCommandError(COMMAND_BYE, USAGE_BYE);
+    }
+
+    /**
+     * Lists all saved tasks.
+     */
+    private static String listCommand(String args) {
+        if (args.equals("")) {
+            if (numSavedTasks == 0) {
+                return MESSAGE_NO_TASKS;
+            }
+            String result = MESSAGE_SHOW_TASKS;
+            for (int i = 0; i < numSavedTasks; i++) {
+                result += LS + (i+1) + ") " + savedTasks[i];
```

Comment:
For this part the expression is a little complicated, maybe you can choose to create a variable that stores these elements and let result be incremented to that?